gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonLongRunningClient.java

```diff
@@ -35,56 +35,108 @@
-  private final PollingRequestFactory<RequestT> pollingRequestFactory;
+  private final UnaryCallable getOperationCallable;
+  private final UnaryCallable cancelOperationCallable;
+  private final UnaryCallable deleteOperationCallable;
 
-  public HttpJsonLongRunningClient(
-      UnaryCallable<RequestT, OperationT> operationCallable,
-      OperationSnapshotFactory<RequestT, OperationT> operationSnapshotFactory,
-      PollingRequestFactory<RequestT> pollingRequestFactory) {
-    this.operationCallable = operationCallable;
-    this.operationSnapshotFactory = operationSnapshotFactory;
-    this.pollingRequestFactory = pollingRequestFactory;
+  private HttpJsonLongRunningClient(
+      UnaryCallable getOperationCallable,
+      UnaryCallable cancelOperationCallable,
+      UnaryCallable deleteOperationCallable) {
+    this.getOperationCallable = getOperationCallable;
+    this.cancelOperationCallable = cancelOperationCallable;
+    this.deleteOperationCallable = deleteOperationCallable;
   }
 
+  //  private final UnaryCallable<RequestT, OperationT> operationCallable;
+  //  private final OperationSnapshotFactory<RequestT, OperationT> operationSnapshotFactory;
+  //  private final PollingRequestFactory<RequestT> pollingRequestFactory;
+
+  //  public HttpJsonLongRunningClient(
+  //      UnaryCallable<RequestT, OperationT> operationCallable,
+  //      OperationSnapshotFactory<RequestT, OperationT> operationSnapshotFactory,
+  //      PollingRequestFactory<RequestT> pollingRequestFactory) {
+  //    this.operationCallable = operationCallable;
+  //    this.operationSnapshotFactory = operationSnapshotFactory;
+  //    this.pollingRequestFactory = pollingRequestFactory;
+  //  }
+
   /** {@inheritDoc} */
   @Override
   public UnaryCallable<String, OperationSnapshot> getOperationCallable() {
     return TranslatingUnaryCallable.create(
-        operationCallable,
-        new ApiFunction<String, RequestT>() {
+        getOperationCallable,
+        new ApiFunction<String, GetOperationRequest>() {
           @Override
-          public RequestT apply(String id) {
-            return pollingRequestFactory.create(id);
+          public GetOperationRequest apply(String request) {
+            return GetOperationRequest.newBuilder().setName(request).build();
           }
         },
-        new ApiFunction<OperationT, OperationSnapshot>() {
+        new ApiFunction<Operation, OperationSnapshot>() {
           @Override
-          public OperationSnapshot apply(OperationT operation) {
-            return operationSnapshotFactory.create(null, operation);
+          public OperationSnapshot apply(Operation operation) {
+            return HttpJsonOperationSnapshot.create(operation);
           }
         });
   }
 
   /** {@inheritDoc} */
   @Override
   public UnaryCallable<String, Void> cancelOperationCallable() {
-    return null;
+    return TranslatingUnaryCallable.create(
```

Comment:
Looks like cancel and delete were never implemented, they were implemented for gRPC, but not sure if it's an oversight while implementing rest or it was never required for gRPC.

Reply:
I'm not sure either -- It doesn't seem like LongRunningClient is exposed public for gRPC. But the OperationCallable does expose a `cancel()` function so I think it would make sense for HttpJson to at least implement that one.